rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+      String str = readNullTerminatedString(segment,
+          textAreaOffset + ByteBuffer.wrap(segment, currOffset, Integer.BYTES).order(order).getInt());
+      interfaces.add(str);
+      currOffset += 24;
+    }
+    
+    for (short[] entry : entries) {
+      String name = readNullTerminatedString(segment, textAreaOffset + entry[3]);
+      Set<AccessType> set = AccessType.getTypeFromString(entry[1]);
+      
+      switch(ElementKind.getKind(entry[2])) {
+        case METHOD:
+          MethodElement mthd = new MethodElement(name, set, segment, currOffset, textAreaOffset, order);
+          currOffset += mthd.size();
+          unit.methods.add(mthd);
+          break;
+        case PROPERTY:          
+            PropertyElement prop = new PropertyElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += prop.size();
+            unit.properties.add(prop);
+            break;
+        case VARIABLE:
+            VariableElement var = new VariableElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += var.size();
+            unit.variables.add(var);
+          break;
+        case TABLE:
+            TableElement tbl = new TableElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += tbl.size();
+            unit.tables.add(tbl);
+          break;
+        case BUFFER:
+            BufferElement buf = new BufferElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += buf.size();
+            unit.buffers.add(buf);
+          break;
+        case QUERY:
+            QueryElement qry = new QueryElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += qry.size();
+          break;
+        case DATASET:
+            DatasetElement ds = new DatasetElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += ds.size();
+          break;
+        case DATASOURCE:
+           DataSourceElement dso = new DataSourceElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += dso.size();
+          break;
+        case EVENT:
+           EventElement evt = new EventElement(name, set, segment, currOffset, textAreaOffset, order);
+            currOffset += evt.size();
+            unit.events.add(evt);
+          break;
+        case UNKNOWN:
+          throw new InvalidRCodeException("Found element kind " + entry[2]);
+      }
+    }
+  }
+
+  private final void processInitialValueSegment(InputStream input, PrintStream out) throws IOException, InvalidRCodeException {
+    byte[] segment = new byte[initialValueSegmentSize];
+    int bytesRead = input.read(segment);
+    if (bytesRead != initialValueSegmentSize) {
+      throw new InvalidRCodeException("Not enough bytes in initial value segment block");
+    }
+    if (out != null) {
+      out.printf("%n**********%INITIAL VALUES%n***********%n");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Not enough arguments. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2275)